cmd/bosun/sched/sched.go

```diff
@@ -588,11 +598,38 @@ func (s *Schedule) action(user, message string, t models.ActionType, st *models.
 			return "", fmt.Errorf("cannot acknowledge closed alert")
 		}
 		st.NeedAck = false
+	case models.ActionCancelClose:
+		found := false
+		for i, a := range st.Actions {
```

Comment:
I'd say cancel **all** pending delay closes. In addition, don't permit a new one if there is already an open one. I could maybe see wanting to create a new close action to push the deadline back or something, but it feels like something that would be an error more often than not, and allowing it opens up a whole bunch of cans of worms.